Sources/ProjectDescription/Dependency.swift

```diff
@@ -11,18 +11,27 @@ public struct Dependency: Codable, Equatable {
     /// Dependecy manager used to retrieve the dependecy
     public let manager: Dependency.Manager
 
-    public init(name: String, requirement: Dependency.Requirement, manager: Dependency.Manager) {
+    /// Set of platforms for  the given dependency
+    public let platforms: Set<Platform>
+
+    public init(name: String,
+                requirement: Dependency.Requirement,
+                manager: Dependency.Manager,
+                platforms: Set<Platform>) {
```

Comment:
Nitpick: I'd rather have this one as a `[Platform]` and convert it into a `Set` internally. That makes the initialization shorter for the user:
`[.iOS]` vs `Set(...: [.iOS])`